app/lib/import_export.rb

```diff
@@ -541,25 +231,67 @@ def get_export_data(patients, data)
     exported_data
   end
 
-  # Extracts patient data values given relevant fields
+  # Extract patient data values given relevant fields
   def extract_patients_details(patients, fields)
-    # perform the following queries in bulk only if requested for better performance
-    if fields.include?(:jurisdiction_name)
-      jurisdiction_names = Hash[Jurisdiction.find(patients.pluck(:jurisdiction_id).uniq).pluck(:id, :name).map { |id, name| [id, name] }]
-    end
-    if fields.include?(:jurisdiction_path)
-      jurisdiction_paths = Hash[Jurisdiction.find(patients.pluck(:jurisdiction_id).uniq).pluck(:id, :path).map { |id, path| [id, path] }]
+    # query jurisdiction paths in bulk if requested
+    if (fields & %i[jurisdiction_path transferred_from transferred_to]).any?
+      jur_ids = patients.pluck(:jurisdiction_id).uniq
+      jur_ids = (jur_ids + patients.pluck(:latest_transfer_from)).uniq if fields.include?(:transferred_from)
+      jurisdiction_paths = Hash[Jurisdiction.find(jur_ids).pluck(:id, :path)]
     end
 
-    patients_transfers = get_patients_transfers(patients) if (fields & %i[transferred_from transferred_to]).any?
-    patients_laboratories = get_patients_laboratories(patients) if (fields & PATIENT_LAB_FIELDS).any?
-    patients_creators = Hash[User.find(patients.pluck(:creator_id)).pluck(:id, :email)] if fields.include?(:creator)
+    # query jurisdiction names in bulk if requested
+    jurisdiction_names = Hash[Jurisdiction.find(patients.pluck(:jurisdiction_id).uniq).pluck(:id, :name)] if fields.include?(:jurisdiction_name)
+
+    # query user emails in bulk if requested
+    patients_creators = Hash[patients.joins('INNER JOIN users ON patients.creator_id = users.id').pluck('users.id', 'users.email')] if fields.include?(:creator)
+
+    # query patients laboratories in bulk if requested
+    if (fields & PATIENT_LAB_FIELDS).any?
+      patients_laboratories = Laboratory.where(patient_id: patients.pluck(:id))
+                                        .order(specimen_collection: :desc)
+                                        .group_by(&:patient_id)
+                                        .transform_values { |v| v.take(2) }
+    end
 
     # construct patient details
     patients_details = []
     patients.each do |patient|
-      # populate requested inherent fields
-      patient_details = extract_incomplete_patient_details(patient, fields)
+      patient_details = {}
+
+      # populate inherent fields
+      (fields & (PATIENT_FIELD_TYPES[:numbers] + PATIENT_FIELD_TYPES[:strings])).each do |field|
+        patient_details[field] = patient[field]&.to_s || ''
```

Comment:
I think I added this while trying to cast everything to a string when trying to find a workaround for fast_excel writing numbers as numbers instead of strings, but this didn't end up solving the problem so I'll remove it

Reply:
fixed here: https://github.com/SaraAlert/SaraAlert/pull/717/commits/1ee73843e04d95e6b6b2f336b0706f2e5962ae73#diff-c34521fcd0fdc2034d456886aae8fd35299fe908ab88d8e49c60d0b4d39ab26bR271